arbnode/transaction_streamer.go

```diff
@@ -988,14 +1019,64 @@ func (s *TransactionStreamer) WriteMessageFromSequencer(
 		BlockHash:       &msgResult.BlockHash,
 	}
 
+	l1Block := msgWithMeta.Message.Header.BlockNumber
+
+	if s.config().SovereignSequencerEnabled && s.espressoClient != nil && wavmio.IsHotShotLive(l1Block) {
+		ctx, cancel := context.WithTimeout(context.Background(), s.config().EspressoTimeout)
+		defer cancel()
+		err := s.SendL2MessageToEspresso(ctx, msgWithMeta)
+		if err != nil {
+			return fmt.Errorf("failed to send message to espresso: %w", err)
+		}
+	}
+
 	if err := s.writeMessages(pos, []arbostypes.MessageWithMetadataAndBlockHash{msgWithBlockHash}, nil); err != nil {
 		return err
 	}
+
 	s.broadcastMessages([]arbostypes.MessageWithMetadataAndBlockHash{msgWithBlockHash}, pos)
 
 	return nil
 }
 
+func (s *TransactionStreamer) SendL2MessageToEspresso(ctx context.Context, msgWithMeta arbostypes.MessageWithMetadata) error {
+
+	tx := espressoTypes.Transaction{
+		Namespace: s.config().EspressoNamespace,
+		Payload:   msgWithMeta.Message.L2msg,
+	}
+
+	err := s.espressoClient.SubmitTransaction(ctx, tx)
+	if err != nil {
+		return fmt.Errorf("failed to submit transaction to espresso: %w", err)
+	}
+
+	ticker := time.NewTicker(s.config().EspressoTxnsPollingInterval)
+	defer ticker.Stop()
+
+	// Poll to check if the transaction is included in the block
```

Comment:
I don't think we need to do that. We can simply assume that the transactions will always be included as long as the `submit` returns 200.

Reply:
Based on our discussion in standup, will create a new thread to handle the queue of batches